datalake/src/main/java/com/sequenceiq/datalake/service/resize/recovery/ResizeRecoveryService.java

```diff
@@ -0,0 +1,116 @@
+package com.sequenceiq.datalake.service.resize.recovery;
+
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.NotImplementedException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.dto.NameOrCrn;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.response.recovery.RecoveryStatus;
+import com.sequenceiq.datalake.entity.DatalakeStatusEnum;
+import com.sequenceiq.datalake.entity.SdxCluster;
+import com.sequenceiq.datalake.entity.SdxStatusEntity;
+import com.sequenceiq.datalake.flow.SdxReactorFlowManager;
+import com.sequenceiq.datalake.flow.chain.DatalakeResizeFlowEventChainFactory;
+import com.sequenceiq.datalake.service.recovery.RecoveryService;
+import com.sequenceiq.datalake.service.sdx.SdxService;
+import com.sequenceiq.datalake.service.sdx.status.SdxStatusService;
+import com.sequenceiq.flow.core.Flow2Handler;
+import com.sequenceiq.flow.domain.FlowLog;
+import com.sequenceiq.flow.service.flowlog.FlowChainLogService;
+import com.sequenceiq.sdx.api.model.SdxRecoverableResponse;
+import com.sequenceiq.sdx.api.model.SdxRecoveryRequest;
+import com.sequenceiq.sdx.api.model.SdxRecoveryResponse;
+
+@Component
+/**
+ * Provides entrypoint for recovery of failed SDX resize.
+ *
+ * The main entry point is {@code triggerRecovery}, which starts a cloudbreak Flow to recover the Data Lake.
+ * To ensure a Resize recovery is appropriate, use one of {@code canRecover} or {@code validateRecovery}.
+ */
+public class ResizeRecoveryService implements RecoveryService {
```

Comment:
resize recovery should be behind an entitlement, since resize feature itself is behind an entitlement too. also I would suggest to use a different entitlement, since it gives the possibility to release the resize feature without recovery first.